### PR TITLE
Spkonfig

### PR DESCRIPTION
**Pull request by alpet**

### CODE REVIEW

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/ApplicationTests.kt [hunk 1]
```diff
@@ -106,6 +107,7 @@ class ApplicationTests {
 	@AfterAll
 	fun stopKafkaConsumers() {
 		kafkaListener.close()
+//		super.close()
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/ApplicationTests.kt:110]: Remove uncommented code

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/admin/AdminInterfaceTests.kt [hunk 2]
```diff
@@ -33,8 +34,8 @@ import java.util.concurrent.TimeUnit
 @ActiveProfiles("test")
 @SpringBootTest
 @DirtiesContext(classMode = DirtiesContext.ClassMode.AFTER_CLASS)
-@Import(ContainerizedKafka::class)
-class AdminInterfaceTests {
+//@Import(ContainerizedKafka::class)
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/admin/AdminInterfaceTests.kt:37]: Remove uncommented code

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt [hunk 3]
```diff
@@ -1,48 +1,90 @@
...
+		@DynamicPropertySource
+		fun properties(registry: DynamicPropertyRegistry) {
+			start()
+			registry.add(
+				"kafka.brokers", kafkaContainer::getBootstrapServers
+			)
+		}
+
+	//	@JvmStatic
+	//	@BeforeAll
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:55]: Remove uncommented code

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/KafkaListener.kt [hunk 4]
```diff
@@ -69,22 +70,22 @@ class KafkaListener(private val kafkaConfig: AppConfiguration.KafkaConfig) {
...
-		it[StreamsConfig.BOOTSTRAP_SERVERS_CONFIG] = kafkaConfig.kafkaBrokers
+		it[StreamsConfig.BOOTSTRAP_SERVERS_CONFIG] = kafkaConfig.brokers
 		it[StreamsConfig.DEFAULT_KEY_SERDE_CLASS_CONFIG] = Serdes.StringSerde::class.java
 		it[StreamsConfig.DEFAULT_VALUE_SERDE_CLASS_CONFIG] = SpecificAvroSerde::class.java
 		it[StreamsConfig.DEFAULT_DESERIALIZATION_EXCEPTION_HANDLER_CLASS_CONFIG] = LogAndContinueExceptionHandler::class.java
 		it[StreamsConfig.COMMIT_INTERVAL_MS_CONFIG] = 1000
 
-		if (kafkaConfig.secure == "TRUE") {
-			it[CommonClientConfigs.SECURITY_PROTOCOL_CONFIG] = kafkaConfig.protocol
+		if (kafkaConfig.security.enabled == "TRUE") {
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/KafkaListener.kt:81]: Is it possible to change the type of `enabled` from String to Boolean?

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt [hunk 5]
```diff
@@ -1,48 +1,90 @@
 package no.nav.soknad.arkivering.soknadsarkiverer.utils
 
-import no.nav.soknad.arkivering.soknadsarkiverer.config.*
+import no.nav.security.token.support.client.spring.ClientConfigurationProperties
+import no.nav.soknad.arkivering.soknadsarkiverer.kafka.KafkaConfig
+import org.apache.commons.compress.harmony.pack200.PackingUtils.config
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:5]: There are some unused imports that I cant resolve. Could you right-click on the file in the list and select Optimize Imports, please? :)

You could also right-click and do a Reformat Code

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt [hunk 6]
```diff
@@ -1,48 +1,90 @@
...
+
+			kafkaConfig = binder.bind("kafka", KafkaConfig::class.java).get()
+		}
 
-	override fun close() {
-		kafkaContainer.stop()
-	}
+
+		// @Container
+		 val kafkaContainer: KafkaContainer = KafkaContainer(DockerImageName.parse("confluentinc/cp-kafka"))
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:42]: Make it private

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt [hunk 7]
```diff
@@ -1,48 +1,90 @@
...
+		private fun start() {
+			kafkaContainer.start()
+
+		listOf(  kafkaConfig.topics.mainTopic, kafkaConfig.topics.processingTopic, kafkaConfig.topics.messageTopic, kafkaConfig.topics.metricsTopic)
+				.forEach { createTopic(it) }
+		}
+
+		@JvmStatic
+		@AfterAll
+		fun close() {
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:65]: Are you sure this method is called? Have you observed the output in the logs? IntelliJ reports it as unused, but the annotations can throw it off sometimes.

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt [hunk 8]
```diff
@@ -1,48 +1,90 @@
...
+	companion object {
+		private val kafkaConfig : KafkaConfig
+
+		init {
+			val factoryBean = YamlPropertiesFactoryBean()
+			factoryBean.setResources(ClassPathResource("application.yml"))
+
+			val properties: Properties? = factoryBean.getObject()
+
+			val propertySource: ConfigurationPropertySource = MapConfigurationPropertySource(properties)
```
Comment (sjoblomjsqueed) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:34]: A bit of a matter of taste, but I'd remove the types. Especially when the type of `properties`  is `Properties?`, it doesn't add so much to explicitly state it, and the downside is extra cognitive load when each line is more packed with extra information.

**Review by sjoblomjsqueed — COMMENTED** [review]
Great work. Just a few minor comments

File: arkiverer/src/main/resources/application.yml [hunk 9]
```diff
@@ -57,7 +79,7 @@ no.nav.security.jwt:
 spring:
   config:
     activate:
-      on-profile: spring | default | docker | dev | prod
+      on-profile: spring | default | dev | prod
```
[the remark this answers is not in the document]
Comment (alpet) [reply arkiverer/src/main/resources/application.yml:82]: Unfortunatly no :) This is one majour underwarter stone, but i will fix it in the next commit.

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/admin/AdminInterfaceTests.kt (hunk 2 above)
Comment (alpet) [reply arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/admin/AdminInterfaceTests.kt:37]: this is intermediate comit. it is not job done yet.

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt (hunk 6 above)
Comment (alpet) [reply arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:42]: This code is not going to servive the next commit.

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt (hunk 7 above)
Comment (alpet) [reply arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:65]: Yep, very much called. https://howtodoinjava.com/junit5/after-all-annotation-example/

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt (hunk 8 above)
Comment (alpet) [reply arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:34]: It was automatic convert from Java :D

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/kafka/bootstrapping/StateRecreationTests.kt [hunk 10]
```diff
@@ -59,6 +60,8 @@ class StateRecreationTests {
 
 	@Autowired
 	private lateinit var appConfiguration: AppConfiguration
+	@Autowired
+	private lateinit var kafkaConfig: KafkaConfig
 	private lateinit var kafkaMainTopicProducer: KafkaProducer<String, Soknadarkivschema>
```
Comment (sturleh) [inline arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/kafka/bootstrapping/StateRecreationTests.kt:65]: missing @autowired?

Comment (alpet) [reply arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/kafka/bootstrapping/StateRecreationTests.kt:65]: Den er initialisert på linje 75. Det er en blanding av Dependency Injection og vanlig Java initialization.

File: arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt [hunk 11]
```diff
@@ -1,48 +1,90 @@
...
+			val binder = Binder(propertySource)
+
+			kafkaConfig = binder.bind("kafka", KafkaConfig::class.java).get()
+		}
 
-	override fun close() {
-		kafkaContainer.stop()
-	}
+
+		// @Container
```
[the remark this answers is not in the document]
Comment (alpet) [reply arkiverer/src/test/kotlin/no/nav/soknad/arkivering/soknadsarkiverer/utils/ContainerizedKafka.kt:37]: This code will be rewritten in a JUnit extension. I am not touching it further for the moment.

**Pull request merged**